submissions/yanyshpolska/OOP-exercise/index.js

```diff
@@ -0,0 +1,78 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: https://github.com/Yanyshpolska/a-tiny-JS-world
+   Web app: https://yanyshpolska.github.io/a-tiny-JS-world/
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+class Inhabitant {
+  constructor(name, saying, sex) {
+    this.name = name;
+    this.saying = saying;
+    this.sex = sex;
+  }
+  getInfo() {
+    return this.name + ", " + this.saying + ", " + this.sex;
+  }
+}
+
+class Human extends Inhabitant {
+  hands = 2;
+  legs = 2;
+  constructor(...args) {
+    super(...args);
+  }
+  getInfo() {
+    return (
+      super.getInfo() +
+      ", " +
+      this.hands +
+      ", " +
+      this.legs +
+      ", " +
+      this.constructor.name
+    );
+  }
+}
+
+class Animal extends Inhabitant {
+  tail = 1;
+  legs = 4;
+  constructor(...args) {
+    super(...args);
+  }
+  getInfo() {
+    return super.getInfo() + ", " + this.legs + ", " + this.tail;
+  }
+}
+
+class Cat extends Animal {
+  constructor(...args) {
+    super(...args);
+  }
+  getInfo() {
+    return super.getInfo() + ", " + this.constructor.name;
```

Comment:
I wonder if we can generalize use of `this.constructor.name`? Too much repetition across the entire codebase.
Did you try to use it in the base class?

Reply:
Reworked as you suggested.